Source/1. Functional Programmingming/js/1_PartialApplication.js

```diff
@@ -0,0 +1,19 @@
+﻿var partialFn = (function () {
+    "use strict";
+    return function () {
+        //convert argumets to array removing last argument
+        var initialArgs = Array.prototype.slice.call(arguments, 0, -1),
+            initialFn = arguments[arguments.length - 1];
+        //check if last argument is function
```

Comment:
I'd suggest avoiding the assumption, that the initial function will be passed as last argument. The better choice is to make it an implicit parameter: Function.prototype.partialFn () { ...